### DISCUSSION TITLE
Table Selection CheckedChanged event?

### DISCUSSION THREAD

**Discussion by michaelcsikos** [ant-design-blazor/ant-design-blazor#2899] [Q&A]
In C# code, how can I respond to the selection being changed?

**Comment by ElderJames** [comment]
Please take a look at this demo: https://antblazor.com/zh-CN/components/table#components-table-demo-blazor

**Reply by michaelcsikos** [reply]
@ElderJames I have looked at all the examples but I can't see any available event. The `Selection` class inner `Checkbox` subscribes to `OnCheckedChange(bool selected)` but it is `private`. It would be good if this was at least `protected virtual` or even better would be another event we could subscribe to which the `Selection` fired.

**Reply by ElderJames** [reply]
You don't want `SelectedItemsChanged` ?

**Reply by michaelcsikos** [reply]
`OnSelectedItemsChanged` is not available in `Table` or the `Selection` column, is it? I thought it was only in `SelectBase`.

**Reply by ElderJames** [reply]
Oh sorry, I meat `SelectedRowsChanged` in Table.

**Reply by michaelcsikos** [reply]
When I try to hook up `SelectedRowsChanged` it says:
`RZ10010 The component parameter 'SelectedRowsChanged' is used two or more times for this component. Parameters must be unique (case-insensitive). The component parameter 'SelectedRowsChanged' is generated by the '@bind-SelectedRows' directive attribute.`

**Reply by ElderJames** [reply]
Yes, so you don't use two-way binding, use `SelectedRows` and `SelectedRowsChanged` separately.

**Reply by michaelcsikos** [reply]
OK, this event doesn't tell me which row was selected or deselected.

**Reply by ElderJames** [reply]
Yes, it will only pass in the currently selected rows. If more information is needed, I think we should add another event

**Reply by michaelcsikos** [reply]
I can use a `Dictionary` to keep a track of what has been added and removed. It's not particularly efficient, but it's a workaround for now.
